airbyte_cdk/test/standard_tests/docker_base.py

```diff
@@ -121,15 +123,29 @@ def get_scenarios(
 
                 scenario = ConnectorTestScenario.model_validate(test)
 
-                if scenario.config_path and scenario.config_path in [
-                    s.config_path for s in test_scenarios
-                ]:
-                    # Skip duplicate scenarios based on config_path
-                    continue
-
                 test_scenarios.append(scenario)
 
-        return test_scenarios
+        # Remove duplicate scenarios based on config_path.
+        deduped_test_scenarios: list[ConnectorTestScenario] = []
+        for scenario in test_scenarios:
+            for existing_scenario in deduped_test_scenarios:
+                if scenario.config_path == existing_scenario.config_path:
+                    # If a scenario with the same config_path already exists, we merge the empty streams.
+                    # scenarios are immutable, so we create a new one.
+                    all_empty_streams = (existing_scenario.empty_streams or []) + (
+                        scenario.empty_streams or []
+                    )
```

Comment:
This implementation looks correct but we may want to dedupe it after creation.


```suggestion
                    all_empty_streams = (existing_scenario.empty_streams or []) + (
                        scenario.empty_streams or []
                    )
                    # Dedupe the list:
                    all_empty_streams = list(set(all_empty_streams))
```